src/App.tsx

```diff
@@ -1,26 +1,177 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import classNames from 'classnames';
+
 import { UserWarning } from './UserWarning';
+import {
+  changeCompletedTodo,
+  deleteTodo,
+  getTodos,
+  USER_ID,
+} from './api/todos';
+
+import { FilterTodo, LoadingItem, Todo } from './types/Todo';
+
+import { ListTodo } from './components/ListTodo';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
 
-const USER_ID = 0;
+import { loadingList } from './utils/loadingList';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [completedTodos, setCompletedTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState(FilterTodo.all);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoadingItems, setIsLoadingItems] = useState<LoadingItem[]>([]);
+
+  useEffect(() => {
+    getTodos()
+      .then(data => {
+        setTodos(data);
+        setTempTodo(null);
+      })
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+        setTimeout(() => setErrorMessage(''), 3000);
+      });
+  }, [tempTodo]);
```

Comment:
You shouldn't make a new request depending on the temporary todo. Consider updating the todo list depending on the response after creating the todo